xdsl/transforms/stencil_shape_minimize.py

```diff
@@ -42,6 +51,27 @@ def match_and_rewrite(self, op: func.FuncOp, rewriter: PatternRewriter, /):
             op.update_function_type()
 
 
+@dataclass
+class RestrictStoreOp(RewritePattern):
+    restrict: tuple[int, ...]
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: stencil.StoreOp, rewriter: PatternRewriter, /):
+        new_bounds = [
+            (min(lower_bound, bound_lim), min(upper_bound, bound_lim))
+            for lower_bound, upper_bound, bound_lim in zip(
+                op.bounds.lb, op.bounds.ub, self.restrict
+            )
+        ]
```

Comment:
nit: An alternative solution that didn't come to mind earlier would be to build (or pass in) a `restrict_bounds` StencilBoundsAttr and use one of its overload, i.e. `op.bounds & restrict_bounds` to build the new bounds. Feel free to ignore, but wanted to mention.

Reply:
noted, thanks for this